offerStatus (#48)
cultuurnet/uit-documentatie: content/json-ld-crud-api/latest/events/event-status.md

---
---

# Event - status

The status indicates whether an event takes place as planned, has been postponed or cancelled. The following values for 'status' are applicable:
* `available`: the event takes place as planned. This is the default value when no status for the event or subEvent is included.
* `temporarilyUnavailable`: the event has been postponed to a later date yet to be determined.
* `unavailable`: the event has been cancelled.

Optionally, a reason for the new status can be added. 

When the status is set back to `available`, the reasons (if present) will be automatically removed.

It is possible to set the status on top event level or on subEvent level:
* **top event level**: for the whole event and all its subEvents (occurences). All subEvents will inherit the status from the top event level

> * **subEvent level**: for a one or multiple subEvent(s). The status on top event level will automtically be derived from the (different) statusses on subEvent level.

Reviewer: ```suggestion
* **subEvent level**: for one or multiple subEvent(s). The status on top event level will be automatically derived from the (different) statuses on subEvent level.
```

## Set status on top event level

It is possible to set the status on top event level for all calendarTypes: `single`, `multiple`, `permanent`, `periodic`. If the event has subEvents, all subEvents of the event will inherit the status (and reason) from the top event level.

**Example**

```
"status":{
   "type":"Unavailable",
   "reason":{
      "nl":"De lessenreeks is omwille van coronamaatregelen geannuleerd",
      "fr":"Le série de cours a été annulé en raison de mesures corona"
   }
}
```

## Set status on subEvent level

For calendarTypes `single` and `multiple` it is also possible to set the status on subEvent level.

> The status on top event level will be derived from the statusses on the subEvent level.

Reviewer: ```suggestion
The status on top event level will be derived from the statuses on the subEvent level.
```

Reviewer: There are a few other occurrences of `statusses`, I think it has to be `statuses`.

When UiTdatabank calculates the status for the top event level, the following logic is being applied:

| status subEvents | Derived status on top event level |
| ------------- | ---------------- | 
| All available | available | 
| All unavailable | unavailable | 
| All temporarilyUnavailable | temporarilyUnavailable | 
| Mixed, but one or more available | available | 

> | Mixed, but one or more temporarilyUnavailable | temporarilyUnavailable | 

Reviewer: ```suggestion
| Mixed, but one or more temporarilyUnavailable and none available | temporarilyUnavailable | 
```

Reviewer 2: I think the idea is that you read the table top to bottom and the first condition that matches is the status that gets derived. So `none available` is implied if you're already past `Mixed, but one or more available`

Reviewer 2: But if that wasn't clear than maybe it's not clear enough and we need to make it more explicit and find another format

Author: Haha, flowcharts ftw. I understand why it could be confusing when you're reading top to bottom. With Luc's suggestion we avoid that 👍

**Example 1**

Event with calendarType `single` (containing one subEvent)

```
"subEvent":[
   {
      "@type":"Event",
      "endDate":"2022-04-01T22:00:00+01:00",
      "startDate":"2022-04-01T20:00:00+01:00",
      "status":{
         "type":"TemporarilyUnavailable",
         "reason":{
            "nl":"Door ziekte van de zanger is het concert van 01/04 uitgesteld",
            "en":"The concert on 01/04 has been postponed due to illness of the lead vocalist."
         }
      }
   }
]
```

> For events with calendarType `single`, the derived status and reason (if applicable) for the top event will always be equal to the status (and reason) of the subEvent. Thus, in the example above the derived status for the top event will become `TemporarilyUnavailable`

Reviewer 2: Are you sure the reason also gets derived? It would make sense for `single` now that I think about it, but I'm not sure we implemented that.

Author: Good remark, it made me doubt. Checked it with an event with calendarType `multiple` and that confirms we implemented it that way :thumbsup:

Reviewer: I also doubt that the reason part of the sub events get applied to the top level. This is how we handle an offer without a top level status https://github.com/cultuurnet/udb3-silex/blob/415daf2d3bad98d06a4569abf29298469e008b40/src/Calendar.php#L97

It gets an empty reason

Reviewer 2: Yeah I double-double-checked with Corneel and he said in Slack that he misunderstood and will change it 👍

Reviewer: Ah but maybe I am wrong because this is about the import and not the API which I pasted here.
